gspread/worksheet.py

```diff
@@ -1375,6 +1397,15 @@ def insert_rows(self, values, row=1, value_input_option=ValueInputOption.raw):
                 "can't insert row in worksheet with colon ':' in its name. See issue: https://issuetracker.google.com/issues/36761154"
             )
 
+        if inherit_from_before and row == 1:
+            raise GSpreadException(
+                "inherit_from_before cannot be used when inserting row(s) at the top of a spreadsheet"
+            )
+        if not inherit_from_before and row == self.row_count + 1:
+            raise GSpreadException(
+                "inherit_from_before must be set to True when inserting row(s) at the bottom of a spreadsheet"
+            )
```

Comment:
this is a very nice catch but we will be facing an old gspread issue:
the value of `self.row_count` might be outdated, this is because when adding rows we don't update the sheet properties. so adding new row will not increment that value, meaning the first row the user can use the flag (`inherit_from_before=True`, then second call  to add a row at the same index, which is not the last one anymore, will fail or it should not :slightly_frowning_face: )

I am still looking for a solution to that issue